test/unit/auth-callback-request.js

```diff
@@ -77,7 +77,8 @@ describe('AuthCallbackRequest', () => {
 
       let oidcManager = {}
       let host = { serverUri: 'https://example.com' }
-      let session = { returnToUrl: 'https://example.com/resource' }
+      let returnToUrl = 'https://example.com/resource#hash'
+      let session = { returnToUrl: encodeURIComponent(returnToUrl) }
```

Comment:
I am assuming (based on what I saw the 401 error handler doing in solid-server) that the `returnToUrl` is being url-encoded before being stuffed in a query parameter. (So, I'm simulating that in this unit test). Is that not the right assumption?

Reply:
Yeah, because it needs to be URL-encoded as an URL parameter. However, decoding is part of URL query string parsing, so it should be fine (and in any case, the session should have the unencoded value).